CONTRIBUTING.md

```diff
@@ -0,0 +1,102 @@
+   commit in a patchset is validated against those rules.
+3. Adhere [How to write a commit message][commit-message] rules. Highlights:
+   - Describe a reason of the change, **why** it is necessary (see rule 8). The
+     real reason. Bad example: 'it reaches end-of-life'. Good example: 'it
+     reaches end-of-life and its support is problematic due to the following
+     problems: X, Y, Z'.
+   - Follow 50/72 rule (see rules 2 and 7). 50 chars length for the title is
+     not a strict requirement, but a good soft limit. 72 chars for the commit
+     body is the strict requirement for a prose text, but it can be overrun for
+     listings such as code or logs.
+4. Look on the code around and mimic the style. If in doubt, consult with the
+   style guides for [C][c-style] and [Lua][lua-style].
+5. Don't hesitate to ask a help. If you have doubts, highlight them for a
+   reviewer in the commit message, the pull request description or pull request
+   comments.
+6. Wait for CI and fix all problems.
+   - Glance on the [checkpatch][checkpatch] output if the CI job is red.
+   - If there are test fails that look irrelevant to the changes, highlight
+     this fact in the pull request comments.
+
+Once you received a review, follow the next suggestions to make the process
+comfortable for everyone:
+
+1. Don't add fixup commits on top of the initial patchset. Squash fixups into
+   appropriate commits and force-push your branch. Your patchset will land to
+   `master` as is, without any squashing or reformatting. Keep it in a good
+   shape.
+2. React on comments and respond with summary of changes. If you disagree with
+   a comment, describe your arguments or doubts.
+3. Work in iterations. Either process all comments at once or mark a pull
+   request as draft and return it back when all comments will be answered. A
+   reviewer always want to just look and say 'everything is nice' rather than
+   request changes and remind about forgotten things.
+4. If you reached out of spare time, mark the pull request as draft or close
+   it.
+
+## Useful links
+
+* [Good first issues][good-first-issues] -- if you search for a task to learn
+  Tarantool's code.
+* [How to self-review][self-review] -- more detailed recommendations how to
+  make nice patches.
+* [How to get involved][get-involved] -- what else goes on around Tarantool and
+  how to get involved.
+
+[ideas]: https://github.com/tarantool/tarantool/discussions/categories/ideas
+[q-a]: https://github.com/tarantool/tarantool/discussions/categories/q-a
+[changelogs]: doc/changelogs.md
+[release-notes]: https://www.tarantool.io/en/doc/latest/contributing/release_notes/
+[docbot]: https://github.com/tarantool/docbot
+[commit-message]: https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message
+[c-style]: https://www.tarantool.io/en/doc/latest/dev_guide/c_style_guide/
+[lua-style]: https://www.tarantool.io/en/doc/latest/dev_guide/lua_style_guide/
```

Comment:
Are these the latest versions of the style guides?
There are some guidelines here, too: https://github.com/tarantool/tarantool/wiki/

Recently we considered moving developer guidelines close to code and away from the website.
I suggest providing links to relevant resources updated by developers. If you do so, I will remove the old versions of the style guides from the website.

Reply:
Those are the latest at the moment. We'll move development information to the repository/wiki, but I would move forward step by step. My goal here was to highlight typical problems for contributors with linking from relevant GitHub pages. Actualization and restructurization of the development information will be the next step.

Reply:
Thanks for the feedback!